e2e/vm/support_bundle_test.go

```diff
@@ -0,0 +1,159 @@
+			entries, err := os.ReadDir(".")
+			gomega.Expect(err).Should(gomega.BeNil())
+			bundleExists := false
+			for _, dirEntry := range entries {
+				if strings.Contains(dirEntry.Name(), "finch-support") {
+					_, err := os.Stat(dirEntry.Name())
+					if err == nil {
+						bundleExists = true
+					}
+
+					reader, err := zip.OpenReader(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+
+					zipBaseName := path.Base(dirEntry.Name())
+					zipPrefix := strings.TrimSuffix(zipBaseName, path.Ext(zipBaseName))
+					_, err = reader.Open(path.Join(zipPrefix, "logs", "serial.log"))
+					gomega.Expect(err).ShouldNot(gomega.BeNil())
+
+					err = os.Remove(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+				}
+			}
+			gomega.Expect(bundleExists).Should(gomega.BeTrue())
+		})
+		ginkgo.It("Should generate a support bundle with the same file included and excluded", func() {
+			includeFilename := fmt.Sprintf("tempTestfile%s", time.Now().Format("20060102150405"))
+			_, err := os.Create(includeFilename)
+			gomega.Expect(err).Should(gomega.BeNil())
+			defer func() {
+				err := os.Remove(includeFilename)
+				gomega.Expect(err).Should(gomega.BeNil())
+			}()
+
+			command.Run(o, "support-bundle", "generate", "--include", includeFilename, "--exclude", includeFilename)
+			entries, err := os.ReadDir(".")
+			gomega.Expect(err).Should(gomega.BeNil())
+			bundleExists := false
+			for _, dirEntry := range entries {
+				if strings.Contains(dirEntry.Name(), "finch-support") {
+					_, err := os.Stat(dirEntry.Name())
+					if err == nil {
+						bundleExists = true
+					}
+
+					reader, err := zip.OpenReader(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+
+					zipBaseName := path.Base(dirEntry.Name())
+					zipPrefix := strings.TrimSuffix(zipBaseName, path.Ext(zipBaseName))
+					_, err = reader.Open(path.Join(zipPrefix, "misc", includeFilename))
+					gomega.Expect(err).ShouldNot(gomega.BeNil())
+
+					err = os.Remove(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+				}
+			}
+			gomega.Expect(bundleExists).Should(gomega.BeTrue())
+		})
+		ginkgo.It("Should fail to generate a support bundle when the VM is nonexistent", func() {
+			command.New(o, "vm", "stop").WithTimeoutInSeconds(90).Run()
+			command.New(o, "vm", "remove").WithTimeoutInSeconds(60).Run()
+
+			command.New(o, "support-bundle", "generate").WithoutSuccessfulExit().Run()
+
+			entries, err := os.ReadDir(".")
+			gomega.Expect(err).Should(gomega.BeNil())
+			bundleExists := false
+			for _, dirEntry := range entries {
+				if strings.Contains(dirEntry.Name(), "finch-support") {
+					_, err := os.Stat(dirEntry.Name())
+					if err == nil {
+						bundleExists = true
+					}
+
+					err = os.Remove(dirEntry.Name())
+					gomega.Expect(err).Should(gomega.BeNil())
+				}
+			}
+			gomega.Expect(bundleExists).Should(gomega.BeFalse())
+
+			command.New(o, "vm", "init").WithTimeoutInSeconds(600).Run()
```

Comment:
Should we move to line 137 and add defer, to succeed other tests even if this fails?